### DISCUSSION TITLE
Pushnotifications and FCM - do I need firebase module?

### DISCUSSION THREAD

**Discussion by code-by** [invertase/react-native-firebase#6131] [Q&A]
I'd like to generate local push notifications using some lib, e.g. 'wix/react-native-notifications' and also receive and display notifications receiving from FCM also. Do I need install @react-native-firebase also so my App will able get notifications from FCM or 'wix/react-native-notifications' will be enough?

**Comment by mikehardy** [comment]
Hi there! I find your use of the words "push notifications" to be imprecise so I'm not sure what you mean.
Can you rephrase your question using these precisely defined terms?

- "FCM" or "APNS" - what most would call a push notification, this is a JSON payload delivered over the network to the device. FCM goes directly to Android devices, and it is sent to the APNS servers for Apple, where Apple then delivers the payload to the device
- "local notification" - this is a visible notification on a device, users may interact with it

Finally, as you rephrase your question, please state your exact use cases. Like:

- I want to use the firebase-admin-sdk in a cloud function (or a bash script using the FCM REST API) to...
- generate an FCM containing (exact JSON payload here) that will be delivered to a collection of devices and...
- when the devices receive the FCM, I want them to...
- ...display a local notification or...process data from the payload in the background...or...?

**Reply by code-by** [reply]
Our server use `https://fcm.googleapis.com/fcm/send` endpoint to deliver FCM to our App. Our App now in stage migrating from RN 0.60.5 to 0.67.X and previously it use `react-native-firebase` v.5 for receiving FCM from firebase and then display push notification. Current version of `react-native-firebase` suggest use 3-party for displaying and user interaction with push notifications, e.g. I could use `wix/react-native-notifications` or `react-native-push-notification`. With latest I can obtain push notification token and send test FCM from firebase console. But App now not receiving FCM generated by our server, and I'm think this is cause lack of `@react-native-firebase/messaging`. So is it required to be used in the App to receive FCM or there something wrong with our backend-server, config, etc.? Thank you

**Reply by mikehardy** [reply]
You didn't really answer all my questions, I would still be guessing really, by answering.
I'm not sure how the other packages work, Invertase maintains Notifee though
You might like https://notifee.app/react-native/docs/integrations/fcm
